molecule/verifier/lint/precommit.py

```diff
@@ -0,0 +1,143 @@
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+
+import os
+
+import sh
+
+from molecule import logger
+from molecule import util
+from molecule.verifier.lint import base
+
+LOG = logger.get_logger(__name__)
+
+
+class PreCommit(base.Base):
+    """
+    `PreCommit`_ is not the default verifier linter.
+
+    `PreCommit`_ is a linter for python files and more.
+
+    Additional options can be passed to ``pre-commit`` through the options
+    dict.  Any option set in this section will override the defaults.
+
+    .. code-block:: yaml
+
+        verifier:
+          name: testinfra
+          lint:
+            name: pre-commit
+            options:
+              remove-tabs:
+
+    Test file linting can be disabled by setting ``enabled`` to False.
+
+    .. code-block:: yaml
+
+        verifier:
+          name: testinfra
+          lint:
+            name: pre-commit
+            enabled: False
+
+    Environment variables can be passed to lint.
+
+    .. code-block:: yaml
+
+        verifier:
+          name: testinfra
+          lint:
+            name: pre-commit
+            env:
+              FOO: bar
+
+    .. _`Pre-Commit`: https://pre-commit.com/
+    """
+
+    def __init__(self, config):
+        """
+        Sets up the requirements to execute `pre-commit` and returns None.
+
+        :param config: An instance of a Molecule config.
+        :return: None
+        """
+        super(PreCommit, self).__init__(config)
+        self._precommit_command = None
+        if config:
+            self._tests = self._get_tests()
+
+    @property
+    def default_options(self):
+        return {}
+
+    @property
+    def default_env(self):
+        return util.merge_dicts(os.environ.copy(), self._config.env)
+
+    def bake(self):
+        """
+        Bake a `pre-commit` command so it's ready to execute and returns None.
+
+        :return: None
+        """
+        self._precommit_command = sh.Command('pre-commit').bake(
+            'run',
+            self.options,
+            '--files',
+            self._tests,
+            _env=self.env,
+            _out=LOG.out,
+            _err=LOG.error)
+
+    def execute(self):
+        if not self.enabled:
+            msg = 'Skipping, verifier_lint is disabled.'
+            LOG.warn(msg)
+            return
+
+        if not len(self._tests) > 0:
```

Comment:
This is unpythonic. I cannot even say what this check does right away with the double negation... The proper way to do this is
```suggestion
        if not self._tests:
```